server/controllers/conversation.js

```diff
@@ -0,0 +1,100 @@
+const asyncHandler = require("express-async-handler");
+const Conversation = require("../models/Conversation");
+const Profile = require("../models/Profile");
+
+const {
+  queryConversationsByProfile,
+  formatConversationFields,
+  populateOtherUser,
+  populateLastMessage,
+} = require("../utils/conversationQueries");
+
+// @route GET /conversations
+// @desc retrieves all conversations from the current logged in user.
+// @access Private
+exports.getAllConversations = asyncHandler(async (req, res) => {
+  const { user } = req;
+
+  const profile = await Profile.findOne({ userId: user.id }).exec();
```

Comment:
Alternatively, we can query the db for the logged in user's profile within the `protect` middleware

Reply:
That is a very fair statement :+1: , I'll add that change insted :smile: